pkg/config/node_test.go

```diff
@@ -338,6 +343,145 @@ func deleteTempResolvConf() {
 	os.Remove("/tmp/resolvConf")
 }
 
+var (
+	installConfig = &types.InstallConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: types.InstallConfigVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		BaseDomain: "cluster.openshift.com",
+		Publish:    types.ExternalPublishingStrategy,
+	}
+	baremetalPlatform = baremetal.Platform{
+		LibvirtURI:                   "qemu+tcp://192.168.122.1/system",
+		ProvisioningNetworkInterface: "ens3",
+	}
+	gcpPlatform = gcp.Platform{
+		ProjectID: "test-project",
+		Region:    "us-east-1",
+	}
+	nonePlatform = none.Platform{}
+)
+
+func gcpInstallConfig() *types.InstallConfig {
+	installConfig.Platform = types.Platform{
+		GCP: &gcpPlatform,
+	}
+	return installConfig
+}
+
+func baremetalInstallConfig() *types.InstallConfig {
+	installConfig.Platform = types.Platform{
+		BareMetal: &baremetalPlatform,
+	}
+	return installConfig
+}
+
+func noneInstallConfig() *types.InstallConfig {
+	installConfig.Platform = types.Platform{
+		None: &nonePlatform,
+	}
+	return installConfig
+}
+
+func createTempInstallConfig(installConfig *types.InstallConfig) {
+	icData, _ := yaml.Marshal(installConfig)
+
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-config-v1",
+			Namespace: "kube-system",
+		},
+		Data: map[string]string{
+			"install-config": string(icData),
+		},
+	}
+
+	controllerConfig, _ := yaml.Marshal(cm)
+	os.WriteFile("/tmp/install-config.yaml", controllerConfig, 0644)
```

Comment:
Let's use `os.MkdirTemp()` instead of hardcoding `/tmp`. It won't necessary always be `/tmp` so tests will not work if ran in such an environment

Reply:
Done.